File: Repainting.java
class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0, pos = 0;
        for(int i = 0; i < section.length; ++i) {
            int sec = section[i];
            if(sec <= pos) continue;
            pos = sec + m - 1;
            ++answer;
        }
        return answer;
    }
}

File: Repainting.c
#include <stdio.h>
#include <stdlib.h>

int compare(const void* a, const void* b) {
    return *(int*)a - *(int*)b;
}

int solution(int n, int m, int* section, int section_len) {
    int answer = 0;
    int start = 0;
    
    qsort(section, section_len, sizeof(int), compare);
    
    while (start < section_len) {
        int end = start;
        while (end < section_len && section[end] - section[start] + 1 <= m) {
            end++;
        }
        start = end;
        answer++;
    }
    
    return answer;
}